discovery-infra/test_infra/helper_classes/hypershift.py

```diff
@@ -71,16 +71,18 @@ def set_nodepool_node_count(self, kube_api_client: ApiClient, node_count: int) -
             body=body,
         )
 
-    def get_nodes(self) -> V1NodeList:
+    def get_nodes(self, ready: bool = False) -> V1NodeList:
         if self.hypershift_cluster_client is None:
             hypershift_cluter_kubeapi_client = create_kube_api_client(self.kubeconfig_path)
             self.hypershift_cluster_client = CoreV1Api(hypershift_cluter_kubeapi_client)
+        nodes = self.hypershift_cluster_client.list_node()
+        if ready:
+            return filterNodeByReadyStatus(nodes)
+        return nodes
 
-        return self.hypershift_cluster_client.list_node()
-
-    def wait_for_nodes(self, node_count: int) -> V1NodeList:
+    def wait_for_nodes(self, node_count: int, ready: bool) -> V1NodeList:
```

Comment:
i think better to have it as wait_for_nodes_in_status and give status no? or it is complicated?

Reply:
I tried that, it's useless
the only status anyone cares about is ready